src/world/script_api/7E2AA0.c

```diff
@@ -1,31 +1,45 @@
 #include "common.h"
 
-typedef struct {
-    /* 0x00 */ s32 unk_00; // door state? 01 = using door
+void set_current_item_entity_render_group(s32);
+s32 get_current_item_entity_render_group(void);
+void func_800EF414(s32, s32);
+void get_flat_collider_normal(s32, f32*, f32*, f32*);
+
+extern EvtScript D_802847D4_7E5654;
+extern EvtScript D_80284F80_7E5E00;
+
+enum DoorScriptType {
+    DOOR_SCRIPT_OPEN_DOOR           = 0,
+    DOOR_SCRIPT_MOVE_WALLS          = 1,
+    DOOR_SCRIPT_DROP_DOOR           = 2,
+    DOOR_SCRIPT_TOGGLE_VIS          = 3,
+};
+
+typedef enum DoorState {
+    DOOR_STATE_0                    = 0,
+} DoorState;
+
+typedef struct AdvancedDoor {
+    /* 0x00 */ DoorState state;
     /* 0x04 */ s32 unk_04;
-    /* 0x08 */ s16 unk_08;
-    /* 0x0A */ s16 unk_0A;
-    /* 0x0C */ f32 unk_0C; // x pos entry ?
-    /* 0x10 */ f32 unk_10;
-    /* 0x14 */ f32 unk_14;
-    /* 0x18 */ f32 unk_18; // rotation entry
-    /* 0x1C */ f32 unk_1C;
-    /* 0x20 */ f32 unk_20;
-    /* 0x24 */ f32 unk_24; // x pos exit ?
-    /* 0x28 */ f32 unk_28; // rotation exit
-    /* 0x2C */ s32* unk_2C; // ptr to script
-    /* 0x30 */ s32* unk_30; // ptr to script
-    /* 0x34 */ s32* unk_34; // ptr to ??? (usually null?)
-    /* 0x38 */ s32* unk_38; // ptr to script
-    /* 0x3C */ s32 unk_3C; //flags (has to do with hiding certain objects)
-    /* 0x40 */ char unk_40[12];
-    /* 0x4C */ s32* unk_4C;
-} DoorStuff;
+    /* 0x08 */ s16 newItemVisGroup;
+    /* 0x0A */ s16 prevItemVisGroup;
+    /* 0x0C */ Vec2XZf unk_0C;
+    /* 0x14 */ Vec2XZf unk_14;
+    /* 0x1C */ Vec2XZf unk_1C;
+	/* 0x24 */ Vec2XZf unk_24;
+	/* 0x2C */ EvtScript* scripts[4];
+	/* 0x3C */ s32 modelID;
+	/* 0x40 */ s32 bgColor[3];
+	/* 0x4C */ s32 scriptIDs[4];
+	/* 0x5C */ s32* npcList;
+	/* 0x60 */ s32 colliderID;
+} AdvancedDoor; // size = 0x64
 
 ApiStatus func_80281C20(Evt* script, s32 isInitialCall) {
-    Bytecode* thisPos = script->ptrReadPos;
+    Bytecode* args = script->ptrReadPos;
```

Comment:
lol wow, I haven't named one of those `thisPos` in AGES. this must be a dinosaur decomp

Reply:
Yeah, world and evt api were probably some of the first ones to be done.